File: Python/백준/Bronze/5585. 거스름돈/거스름돈.py
money = 1000 - int(input())
count = 0

if money-500 >= 0:
    count += 1
    money -= 500
if money-100 >= 0:
    count += money//100
    money = money%100
if money-50 >= 0:
    count += money//50
    money = money%50
if money-10 >= 0:
    count += money//10
    money = money%10
if money-5 >= 0:
    count += money//5
    money = money%5
count += money
print(count)

File: Java/백준/Bronze/5585. 거스름돈/거스름돈.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = 1000 - Integer.parseInt(br.readLine());
        int[] coin  = {500, 100, 50, 10, 5, 1};
        int count = 0;

        for (int c : coin) {
            count += n / c;
            n %= c;
        }
        System.out.println(count);
    }
}